File: cpp/src/ui.cpp
#include "ui.hpp"
#include <iostream>
#include <algorithm>

namespace OffsetsFinder {

void PrintHeader() {
    std::cout << Color::BrightCyan << "╔═══════════════════════════════════════════════════╗" << Color::Reset << "\n";
    std::cout << Color::BrightCyan << Color::Bold << "║     Featured OFFSETS FINDER - C++ EDITION       ║" << Color::Reset << "\n";
    std::cout << Color::BrightCyan << "║              v2.0 - Modular Edition               ║" << Color::Reset << "\n";
    std::cout << Color::BrightCyan << "╚═══════════════════════════════════════════════════╝" << Color::Reset << "\n\n";
}

void PrintGameMenu() {
    std::cout << Color::BrightBlue << Color::Bold << "Select Free Fire variant:" << Color::Reset << "\n";
    std::cout << "  " << Color::BrightYellow << "1." << Color::Reset << " Free Fire (Standard)\n";
    std::cout << "  " << Color::BrightYellow << "2." << Color::Reset << " Free Fire MAX\n";
    std::cout << "  " << Color::BrightYellow << "3." << Color::Reset << " Free Fire TELA\n";
    std::cout << "  " << Color::BrightYellow << "4." << Color::Reset << " Auto-detect from dump\n\n";
}

int GetGameSelection() {
    std::cout << Color::BrightBlue << "Enter your choice (1-4): " << Color::Reset;
    int choice;
    std::cin >> choice;
    std::cin.ignore();
    return choice;
}

void PrintResults(const std::vector<OffsetResult>& results, GameVariant game) {
    std::cout << "\n" << Color::BrightCyan << Color::Bold 
              << "====== " << GameVariantToString(game) << " OFFSETS ======" 
              << Color::Reset << "\n";
    
    std::string currentCategory;
    for (const auto& result : results) {
        if (result.category != currentCategory) {
            std::cout << "\n" << Color::BrightMagenta << Color::Bold 
                      << "--- " << result.category << " ---" 
                      << Color::Reset << "\n";
            currentCategory = result.category;
        }
        
        if (result.offset.has_value()) {
            std::cout << Color::BrightGreen << Color::Bold << result.name << Color::Reset 
                      << " " << Color::BrightYellow << result.offset.value() << Color::Reset << "\n";
        } else {
            std::cout << Color::BrightRed << Color::Bold << result.name << Color::Reset 
                      << " " << Color::BrightRed << "NOT FOUND" << Color::Reset << "\n";
        }
    }
}

void PrintStatistics(const std::vector<OffsetResult>& results) {
    size_t found = std::count_if(results.begin(), results.end(),
        [](const auto& r) { return r.offset.has_value(); });
    size_t total = results.size();
    size_t missing = total - found;
    
    std::cout << "\n" << Color::BrightCyan << "═══════════════════════════════════════════" << Color::Reset << "\n";
    std::cout << Color::BrightCyan << Color::Bold << "STATISTICS" << Color::Reset << "\n";
    std::cout << Color::BrightCyan << "═══════════════════════════════════════════" << Color::Reset << "\n";
    std::cout << Color::BrightWhite << "Total offsets: " << Color::Reset 
              << Color::BrightYellow << Color::Bold << total << Color::Reset << "\n";
    std::cout << Color::BrightGreen << Color::Bold << "Found: " << Color::Reset 
              << Color::BrightGreen << Color::Bold << found << Color::Reset << "\n";
    std::cout << Color::BrightRed << Color::Bold << "Missing: " << Color::Reset 
              << Color::BrightRed << Color::Bold << missing << Color::Reset << "\n";
    
    float percentage = (static_cast<float>(found) / total) * 100.0f;
    std::cout << Color::BrightWhite << "Success rate: " << Color::Reset 
              << Color::BrightYellow << Color::Bold << std::fixed << std::setprecision(1) 
              << percentage << "%" << Color::Reset << "\n";
    std::cout << Color::BrightCyan << "═══════════════════════════════════════════" << Color::Reset << "\n";
}

void PrintExportMenu() {
    std::cout << "\n" << Color::BrightBlue << Color::Bold << "Export options:" << Color::Reset << "\n";
    std::cout << "  " << Color::BrightYellow << "1." << Color::Reset << " JSON format\n";
    std::cout << "  " << Color::BrightYellow << "2." << Color::Reset << " C++ header (.hpp)\n";
    std::cout << "  " << Color::BrightYellow << "3." << Color::Reset << " Rust module (.rs)\n";
    std::cout << "  " << Color::BrightYellow << "4." << Color::Reset << " Plain text (.txt)\n";
    std::cout << "  " << Color::BrightYellow << "5." << Color::Reset << " Skip export\n\n";
}

int GetExportSelection() {
    std::cout << Color::BrightBlue << "Enter your choice (1-5): " << Color::Reset;
    int choice;
    std::cin >> choice;
    std::cin.ignore();
    return choice;
}

void PrintSuccess(const std::string& message) {
    std::cout << Color::BrightGreen << Color::Bold << "✓ " << Color::Reset 
              << Color::BrightWhite << message << Color::Reset << "\n";
}

void PrintError(const std::string& message) {
    std::cerr << Color::BrightRed << Color::Bold << "✗ " << Color::Reset 
              << Color::BrightRed << message << Color::Reset << "\n";
}

void PrintInfo(const std::string& message) {
    std::cout << Color::BrightBlue << Color::Bold << "ℹ " << Color::Reset 
              << Color::BrightWhite << message << Color::Reset << "\n";
}

void WaitForEnter() {
    std::cout << "\n" << Color::BrightBlue << "Press Enter to exit..." << Color::Reset;
    std::cin.get();
}

}

File: src/ui.rs
use crate::models::{GameVariant, OffsetResult};
use colored::*;
use std::io::{self, Write};

pub fn print_header() {
    println!("{}", "╔═══════════════════════════════════════════════════╗".bright_cyan());
    println!("{}", "║     Featured OFFSETS FINDER - RUST EDITION      ║".bright_cyan().bold());
    println!("{}", "║              v1.0 - Modular Edition               ║".bright_cyan());
    println!("{}", "╚═══════════════════════════════════════════════════╝".bright_cyan());
    println!();
}

pub fn print_game_menu() {
    println!("{}", "Select Free Fire variant:".bright_blue().bold());
    println!("  {} Free Fire (Standard)", "1.".bright_yellow());
    println!("  {} Free Fire MAX", "2.".bright_yellow());
    println!("  {} Free Fire TELA", "3.".bright_yellow());
    println!("  {} Auto-detect from dump", "4.".bright_yellow());
    println!();
}

pub fn get_game_selection() -> Option<usize> {
    print!("{}", "Enter your choice (1-4): ".bright_blue());
    io::stdout().flush().unwrap();
    
    let mut input = String::new();
    io::stdin().read_line(&mut input).ok()?;
    
    input.trim().parse::<usize>().ok()
}

pub fn print_results(results: &[OffsetResult], game: GameVariant) {
    println!("\n{}", format!("====== {} OFFSETS ======", game.name().to_uppercase()).bright_cyan().bold());
    
    let mut current_category = String::new();
    
    for result in results {
        if result.category != current_category {
            println!("\n{}", format!("--- {} ---", result.category).bright_magenta().bold());
            current_category = result.category.clone();
        }
        
        match &result.offset {
            Some(offset) => {
                println!("{} {}", 
                    result.name.bright_green().bold(), 
                    offset.bright_yellow()
                );
            }
            None => {
                println!("{} {}", 
                    result.name.bright_red().bold(), 
                    "NOT FOUND".red()
                );
            }
        }
    }
}

pub fn print_statistics(results: &[OffsetResult]) {
    let found_count = results.iter().filter(|r| r.offset.is_some()).count();
    let total_count = results.len();
    let missing_count = total_count - found_count;
    
    println!("\n{}", "═══════════════════════════════════════════".bright_cyan());
    println!("{}", "STATISTICS".bright_cyan().bold());
    println!("{}", "═══════════════════════════════════════════".bright_cyan());
    println!("{} {}", 
        "Total offsets:".bright_white(), 
        total_count.to_string().bright_yellow().bold()
    );
    println!("{} {}", 
        "Found:".bright_green().bold(),
        found_count.to_string().bright_green().bold()
    );
    println!("{} {}", 
        "Missing:".bright_red().bold(),
        missing_count.to_string().bright_red().bold()
    );
    
    let percentage = (found_count as f32 / total_count as f32) * 100.0;
    println!("{} {:.1}%", 
        "Success rate:".bright_white(),
        percentage.to_string().bright_yellow().bold()
    );
    println!("{}", "═══════════════════════════════════════════".bright_cyan());
}

pub fn print_export_menu() {
    println!("\n{}", "Export options:".bright_blue().bold());
    println!("  {} JSON format", "1.".bright_yellow());
    println!("  {} C++ header (.hpp)", "2.".bright_yellow());
    println!("  {} Rust module (.rs)", "3.".bright_yellow());
    println!("  {} Plain text (.txt)", "4.".bright_yellow());
    println!("  {} Skip export", "5.".bright_yellow());
    println!();
}

pub fn get_export_selection() -> Option<usize> {
    print!("{}", "Enter your choice (1-5): ".bright_blue());
    io::stdout().flush().unwrap();
    
    let mut input = String::new();
    io::stdin().read_line(&mut input).ok()?;
    
    input.trim().parse::<usize>().ok()
}

pub fn print_success(message: &str) {
    println!("{} {}", "✓".bright_green().bold(), message.bright_white());
}

pub fn print_error(message: &str) {
    eprintln!("{} {}", "✗".bright_red().bold(), message.bright_red());
}

pub fn print_info(message: &str) {
    println!("{} {}", "ℹ".bright_blue().bold(), message.bright_white());
}

pub fn wait_for_enter() {
    print!("\n{}", "Press Enter to exit...".bright_blue());
    io::stdout().flush().unwrap();
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
}
